### ISSUE TITLE
bug(react): enter page again will show a blank page. 

### ISSUE THREAD

**Issue by david2tdw** [ionic-team/ionic#19750]
# Bug Report

**Ionic version:**

[x]4.11.2

**Current behavior:**

it will show a blank page at step 7.

**Expected behavior:**

it should show the content in NewItem page.

**Steps to reproduce:**
1. just create a new app follow the instruction:
https://ionicframework.com/docs/react/your-first-app
2. change the codes as described on above page.
3. run server.
4. click the plus button, navigate to the NewItem page.
5. refresh the NewItem page.
6. click the back arrow back to the home page.
7. click the plus button again.

**Comment by OoDeLally** [comment]
I have the exact same problem here.

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you clarify your steps to reproduce? I am seeing the following behavior:

1. Click plus button, navigate to the NewItem page.
2. Refresh the page.
3. Click the `ion-back-button`.

After that, I am not redirected to the home page (The back button probably shouldn't be showing up, but that's a separate issue).

If I set the `defaultHref` on `ion-back-button` to `"/"` and try those steps again, I am brought back to the main page. From there, clicking on the plus button correctly brings me to the NewItem page again.

**Comment by OoDeLally** [comment]
@liamdebeasi Do exactly what @david2tdw suggested.
After the est

**Comment by liamdebeasi** [comment]
I am doing that, but I am not able to reproduce the issue you are describing. I am working off this repo: https://github.com/liamdebeasi/ionic-react-bug

**Comment by OoDeLally** [comment]
In `NewItem.tsx`, replace
```
<IonBackButton />
```
by
```
<IonBackButton defaultHref="/home" />
```

And try again. I could reproduce it in your repo after making this modification.

**Comment by liamdebeasi** [comment]
Oh interesting, locally I was doing `defaultHref="/"` which still sends me to the main page, but doesn't reproduce the bug. I can reproduce the bug now using `defaultHref="/home"`. We will look into a fix for this. Thanks!

**Comment by aissa-bouguern** [comment]
I have the same issue here!

**Comment by aykamko** [comment]
Here's a React hook that provides workaround for now.
```typescript
/**
 * HACK: Prevents an IonPage from rendering blank when it re-enters.
 * Source: https://github.com/ionic-team/ionic/issues/19750
 */
export const useFixIonPageReentry = () => {
  const pageRef = React.useRef<any>();
  useIonViewWillEnter(() => {
    const pageElem = pageRef.current && pageRef.current.ref.current;
    if (pageElem) {
      pageElem.style.removeProperty("display");
    }
  });

  return pageRef;
};
```

Use it in your page component like so:
```typescript
const CoolPage: React.FC = () => {
  const pageRef = useFixIonPageReentry();
  
  return (
   <IonPage ref={pageRef}>
     {...}
   </IonPage>
  );
};
```

Hope this helps!

**Comment by elylucas** [comment]
Hello, this issue should be fixed in 4.11.3. Please give it a try and if you find more issues, comment here and we can reopen this issue.
